### PR TITLE
cmake: update to 3.19.0

### PR DESCRIPTION
**Pull request by ndowens**

### CODE REVIEW

File: srcpkgs/cmake-bootstrap/patches/only-want-half-baked-cmake.patch [hunk 1]
```diff
@@ -1,122 +0,0 @@
...
- #endif
-   cmSystemToolsCTestCommand =
-     cmStrCat(exe_dir, "/ctest", cmSystemTools::GetExecutableExtension());
-@@ -2010,7 +2010,6 @@ void cmSystemTools::FindCMakeResources(const char* argv0)
-     cmSystemToolsCMClDepsCommand.clear();
-   }
- 
--#ifndef CMAKE_BOOTSTRAP
-   // Install tree has
-   // - "<prefix><CMAKE_BIN_DIR>/cmake"
```
Comment (sgn) [inline srcpkgs/cmake-bootstrap/patches/only-want-half-baked-cmake.patch:20]: Did you run the full bootstrap with this change?

Comment (ndowens) [reply srcpkgs/cmake-bootstrap/patches/only-want-half-baked-cmake.patch:20]: If it non-cross-build it builds fine; Currently I am trying to fix cross, then test cmake and cmake-gui against it.

Comment (sgn) [reply srcpkgs/cmake-bootstrap/patches/only-want-half-baked-cmake.patch:20]: Why does it have problem with cross build? The binaries should never run at all.
And why do you remove this patch (to C++ code).

Comment (ndowens) [reply srcpkgs/cmake-bootstrap/patches/only-want-half-baked-cmake.patch:20]: Let me try and remake the patch since it fails to apply on this version

**Comment by ndowens** [comment]
Work on it more after a little while or tomorrow

**Comment by ndowens** [comment]
Had to write a patch to disable TRY_RUN or cross-builds will fail. Resulting cmake binary work as expected.

File: srcpkgs/cmake-bootstrap/patches/0001-We-only-want-a-half-baked-cmake.patch [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
+ Source/cmSystemTools.cxx |  7 +----
+ bootstrap                | 67 ++--------------------------------------
+ 2 files changed, 3 insertions(+), 71 deletions(-)
+
 diff --git Source/cmSystemTools.cxx Source/cmSystemTools.cxx
-index 1e78d36..a678122 100644
+index 97440d2..747cedc 100644
 --- Source/cmSystemTools.cxx
 +++ Source/cmSystemTools.cxx
-@@ -1988,7 +1988,7 @@ void cmSystemTools::FindCMakeResources(const char* argv0)
```
Comment (sgn) [inline srcpkgs/cmake-bootstrap/patches/0001-We-only-want-a-half-baked-cmake.patch:5]: Too much noise in this patch! And you're not the author ;)

**Comment by sgn** [comment]
> Had to write a patch to disable TRY_RUN or cross-builds will fail. Resulting cmake binary work as expected.

Basically, rebase my patch?

File: srcpkgs/cmake-bootstrap/patches/0001-We-only-want-a-half-baked-cmake.patch (hunk 2 above)
Comment (ndowens) [reply srcpkgs/cmake-bootstrap/patches/0001-We-only-want-a-half-baked-cmake.patch:5]: Ok well you can update cmake-bootstrap if you like

**Comment by ndowens** [comment]
> > Had to write a patch to disable TRY_RUN or cross-builds will fail. Resulting cmake binary work as expected.
> 
> Basically, rebase my patch?

I wrote the smaller ones in cmake and cmake-gui

**Pull request closed without merging**